secscanner2junit/vulnerability.py

```diff
@@ -0,0 +1,189 @@
+        except KeyError:
+            return ''
+
+    def _parse_required_property(self, property_name):
+        try:
+            return self._raw_object[property_name]
+        except KeyError:
+            warn("Could not find required property: " + property_name + " in: " + str(self._raw_object))
+            return ''
+
+    def _parse_simple_property_list(self, property_name, property_type):
+        try:
+            elements = self._raw_object[property_name]
+            result = []
+            for element in elements:
+                result.append(property_type(element))
+            return result
+        except KeyError:
+            return []
+
+    def _parse_required_property_list(self, property_name, property_type):
+        try:
+            elements = self._raw_object[property_name]
+            result = []
+            for element in elements:
+                result.append(property_type(element))
+            return result
+        except KeyError:
+            warn("Could not find required property: " + property_name + " in: " + str(self._raw_object))
+            return []
+
+
+# https://gitlab.com/gitlab-org/security-products/security-report-schemas/-/blob/master/src/security-report-format.json
+class _Vulnerability(_Base):
+    def __init__(self, raw_vulnerability: dict):
+        super().__init__(raw_vulnerability)
+        self.__id = self.__parse_id()
+        self.__name = super()._parse_simple_property('name')
+        self.__description = super()._parse_simple_property('description')
+        self.__severity = super()._parse_simple_property('severity')
+        self.__solution = super()._parse_simple_property('solution')
+        self.__identifiers = super()._parse_required_property_list('identifiers', Identifier)
+        self.__links = super()._parse_simple_property_list('links', Link)
+        # details
+        # tracking
+        # flags
+
+    def __repr__(self):
+        return f"Vulnerability(raw_vulnerability={self._raw_object})"
+
+    def __parse_id(self):
+        try:
+            return self._raw_object['id']
+        except KeyError:
+            warn("Could not find id (use fake instead) for vulnerability: " + str(self._raw_object))
+            return 'fake_id_' + str(randrange(1, 10000000))
+
+    def get_testcase_name(self):
+        id_part = " (ID: " + self.__id + ")"
+        severity_part = " (Severity: " + self.get_severity() + ")"
+        if len(self.__name) > 0:
+            return self.__name + id_part + severity_part
+        else:
+            return self.__identifiers[0].get_name() + id_part + severity_part
+
+    def get_location(self):
+        pass
+
+    def get_description(self):
+        if len(self.__description) > 0:
+            return self.__description
+        else:
+            return "no description"
+
+    def get_failure_type(self):
+        return self.__identifiers[0].get_name()
+
+    def get_output(self):
+        return json.dumps(self._raw_object, indent=4)
```

Comment:
With `json.dumps`, the report (see below) is a bit difficult to read for humans (due to formatting), especially compared to the SAST report.
<img width="769" alt="Screenshot 2022-10-06 at 10 11 31" src="https://user-images.githubusercontent.com/16398152/194277545-5533ac3d-2211-437d-b97c-11676d745f46.png">


I'd suggest, we change the output to yaml, this way the report is easier to read for the human eye :)
```python
import yaml
...
return yaml.dump(self._raw_object, width=1000, ident=4,sort_keys=False)

# Explanation:
# *width* needs to be set, as yaml otherwise inserts linebreaks after 80 characters which messes up the description and the urls
# *sort_keys* is True by default, which moves the description to the top and the links to the middle of the report
```

<img width="765" alt="Screenshot 2022-10-06 at 11 02 10" src="https://user-images.githubusercontent.com/16398152/194278388-3c066b5d-cc9e-4085-971f-340ddb72b862.png">

What do you think?